frontend/lib/src/dataframes/arrowParseUtils.ts

```diff
@@ -32,49 +32,47 @@ import unzip from "lodash/unzip"
 import { isNullOrUndefined } from "@streamlit/lib/src/util/utils"
 
 import {
-  IndexTypeName,
   isRangeIndex,
-  RangeIndex,
-  Type,
+  PandasColumnType,
+  PandasIndexTypeName,
+  PandasRangeIndex,
 } from "./arrowTypeUtils"
 
 /**
- * A row-major grid of DataFrame index header values.
+ * Index data value.
  */
 type IndexValue = Vector | number[]
 
 /**
- * A row-major grid of DataFrame index header values.
+ * A row-major grid of DataFrame index data values.
  */
-export type Index = IndexValue[]
+export type IndexData = IndexValue[]
 
 /**
  * A row-major grid of DataFrame column header values.
  * NOTE: ArrowJS automatically formats the columns in schema, i.e. we always get strings.
  */
-export type Columns = string[][]
+export type ColumnNames = string[][]
```

Comment:
Since you are about to update the comments: maybe this comment could be updated with a quick info why this is a multidimensional array. I suppose it is for nested column names?

Reply:
Yeah, to support multi-level column headers. Added it to the comment 👍